pythonforandroid/recipe.py

```diff
@@ -1127,6 +1128,193 @@ def get_recipe_env(self, arch, with_flags_in_cc=True):
         return env
 
 
+class RustCompiledComponentsRecipe(PythonRecipe):
+
+    # Rust toolchain codes
+    # https://doc.rust-lang.org/nightly/rustc/platform-support.html
+    RUST_ARCH_CODES = {
+        "arm64-v8a": "aarch64-linux-android",
+        "armeabi-v7a": "armv7-linux-androideabi",
+        "x86_64": "x86_64-linux-android",
+        "x86": "i686-linux-android",
+    }
+
+    # Build python wheel using `maturin` instead
+    # of default `python -m build [...]`
+    # Note: This option requires `maturin` to be installed in host system
+    use_maturin = False
+
+    # Directory where to find built wheel
+    # For normal build: "dist/*-linux_*.whl"
+    # For maturin: "target/wheels/*-linux_*.whl"
+    built_wheel_pattern = None
+
+    # These are unpacked in hostpython's site-packages
+    # before building
+    # Warning: Don't specify arch specific wheels here
+    hostpython_wheels = {}
```

Comment:
Let's conform to `kivy-ios`, to maintain less burden and mainly to avoid using urls for hostpython3 dependencies:

https://github.com/kivy/kivy-ios/blob/3d94a16d2dfe6cc1723eef52f1f354b7e331422d/kivy_ios/toolchain.py#L819-L822

Let me know what do you think about it.

Reply:
Ah yes, then it will also enable to install architetcure specific packages without hassle.

Reply:
@misl6 as pip is not availabe with hostpython, can I use host system's pip like this:
```
pip3 install <some packgest> --target <path to hostpython site packages> --python-version 38  --only-binary=:all: --upgrade
```

Reply:
If we can avoid it, by installing `pip` for the hostpython (and I think is doable with some minor changes), as we do in `kivy-ios`, I would avoid it.

But, if that's the only option, we can make it work.

Reply:
Done no error reported. We can enable pip for hostpython in maybe some next PR.